### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by elsassph**
Break readme.md into different files.

### CODE REVIEW

File: doc/hmr-usage.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Hot Module Replacement
+
+## LiveReload server
+
+The feature is based on the [LiveReload](https://livereload.com) API. Haxe API will
+set a listener for `LiveReloadConnect` and register a "reloader plugin" to handle changes.
+
+All you have to is to rebuild the Haxe-JS application and let LiveReload inform our
```
Comment (kLabz) [inline doc/hmr-usage.md]: All you have to do*

File: doc/hmr-usage.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+The feature is based on the [LiveReload](https://livereload.com) API. Haxe API will
+set a listener for `LiveReloadConnect` and register a "reloader plugin" to handle changes.
+
+All you have to is to rebuild the Haxe-JS application and let LiveReload inform our
+running application to reload some of the JavaScript files.
+
+Stylesheets and static images will be normally live reloaded.
+
+#### Liverloadx
```
Comment (kLabz) [inline doc/hmr-usage.md]: Livereloadx*

File: doc/hmr-usage.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+2. In your NPM `libs.js`, make sure to include (anywhere):
+
+    ```javascript
+    if (process.env.NODE_ENV !== 'production') {
+        // enable React hot-reload (optional)
+        require('haxe-modular');
+    }
+    ```
+
+3. And in your Haxe code, where your main `ReadctDOM.render` happens:
```
Comment (kLabz) [inline doc/hmr-usage.md]: `ReactDOM.render`

File: doc/hmr-usage.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    ```haxe
+    var app = ReactDOM.render(...);
+
+    #if (debug && react_hot)
+    ReactHMR.autoRefresh(app);
+    #end
+    ```
+
+The feature leverages [react-proxy](https://github.com/gaearon/react-proxy/tree/master) and
+needs to be enabled by calling , preferably in your NPM modules bundle.
```
Comment (kLabz) [inline doc/hmr-usage.md]: Is there something missing between "calling" and the coma?

File: doc/npm-setup.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+It hopefully is understandable that we are defining a "registry" of NPM modules.
+
+In the browser we will have a global object `window.$hx_scope.__registry__` used by
+Modular to resolve NPM modules.
+
+It is important to correctly name the keys of the object (eg. `'react'`) to match the
+Haxe require calls (eg. `@:jsRequire('react')`).
+
+**Tip:** there is nothing forcing your to register NPM modules, you can register any
```
Comment (kLabz) [inline doc/npm-setup.md]: forcing you*

File: doc/start.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+
+## Known limitations
+
+### Troubleshooting
+
+Modular recognises a few additional debugging flags:
+
+- `-D modular_dump`: generate an additional `.graph` file to analyse the dependency graph,
+- `-D modular_debugmap`: generate, for each module, an additiona; `.map.html` file
```
Comment (kLabz) [inline doc/start.md]: additional*

File: doc/start.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+### Dynamic instantiation
+
+Modular isn't aware of `@:keep`, and code without explicit reference will be removed.
+If you use `Type.createInstance`, you'll still have to add an explicit link to these
+classes; even a simple `Array` containing class references.
+
+### Magic init
+
+If you don't know what static `__init__` is, don't worry about it!
+([if your're curious](http://old.haxe.org/doc/advanced/magic#initialization-magic))
```
Comment (kLabz) [inline doc/start.md]: you're*

**Comment by elsassph** [comment]
Happy to get fixes if there are typos - it was late

**Comment by kLabz** [comment]
Ok, I just don't know what is missing for this one: https://github.com/elsassph/haxe-modular/pull/42#discussion_r158626076

**Comment by elsassph** [comment]
Hum, I'll have to re-read it later :D I'll do the fixes indicated myself I think.

**Pull request merged**